packages/fe-fpm-writer/src/section/index.ts

```diff
@@ -107,3 +112,27 @@ export function generateCustomSection(basePath: string, customSection: CustomSec
 
     return fs;
 }
+
+/**
+ * Add a custom section to an existing UI5 application.
+ *
+ * @param {string} basePath - the base path
+ * @param {CustomSection} customSection - the custom section configuration
+ * @param {Editor} [fs] - the mem-fs editor instance
+ * @returns {Promise<Editor>} the updated mem-fs editor instance
+ */
+export function generateCustomSection(basePath: string, customSection: CustomSection, fs?: Editor): Editor {
+    return generate(basePath, customSection, fs);
+}
+
+/**
+ * Add a custom sub section to an existing UI5 application.
+ *
+ * @param {string} basePath - the base path
+ * @param {CustomSection} customSection - the custom section configuration
+ * @param {Editor} [fs] - the mem-fs editor instance
+ * @returns {Promise<Editor>} the updated mem-fs editor instance
+ */
+export function generateCustomSubSection(basePath: string, customSection: CustomSubSection, fs?: Editor): Editor {
+    return generate(basePath, customSection, fs);
```

Comment:
Wouldn't it be cleaner to already identify here which template should be used and then hand it to the internal generate function? As result all decision logic is kept out of the common functions.

Reply:
correction 937c281f77dcc401e7dc782bab21367579ab3a5f